app/forms/openid_connect_authorize_form.rb

```diff
@@ -56,6 +56,13 @@ def submit(user, rails_session_id)
     FormResponse.new(success: success, errors: errors.messages, extra: extra_analytics_attributes)
   end
 
+  # Similar to submit, but only helps generate errors
+  def check_submit
```

Comment:
I wouldn't trust a method like that, because we'd already have to know ahead of time before submitting the request was valid, I think it puts too much responsibility on the caller.